test/workspace.jl

```diff
@@ -24,3 +26,33 @@ mutable struct Foo end
 @assert Tuple{Type{LastMain.Foo}} !== Tuple{Type{Main.Foo}}
 """
 run(`$exename --startup-file=no -e $script2`)
+
+# Issue #22101
+mktempdir() do dir
+    withenv("JULIA_DEBUG_LOADING" => nothing) do
+        # We need to ensure that the module does a nontrivial amount of work during precompilation
+        write(joinpath(dir, "Test22101.jl"), """
+            __precompile__()
+            module Test22101
+                export f22101
+                f22101() = collect(1:10)
+                f22101()
+            end
+        """)
+        write(joinpath(dir, "testdriver.jl"), """
+            insert!(LOAD_PATH, 1, "$dir")
+            insert!(Base.LOAD_CACHE_PATH, 1, "$dir")
+            try
+                using Test22101
+                f22101()
+                workspace()
+                using Test22101
+            finally
+                splice!(LOAD_PATH, 1)
+                splice!(Base.LOAD_CACHE_PATH, 1)
+            end
+            exit(isdefined(Main, :f22101) ? 0 : 1)
+        """)
+        @test success(`$(Base.julia_cmd()) --startup-file=no --precompile=yes $(joinpath(dir, "testdriver.jl"))`)
```

Comment:
The default is to swallow all stdio, which is why the windows failure didn't print anything